### PR TITLE
[FIX] keep annotation from data dict even for column of numeric values

### LINKED ISSUE #11: [BUG] Numeric levels are not expanded
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Expected Behavior

in OpenNeuro `ds000144` I have `SADDx_p` encoding presence of social anxiety.
[From the data dictionary](https://openneuro.org/datasets/ds000144/versions/00002/file-display/participants.json) I can see that it is encoded as
- 0: Not Meeting Criteria for SAD at Preschool
- 1: Meeting Criteria for SAD at Preschool

I would like to see both unique values (or any others I haven't seen) in in the .tsv, 
so that I can annotate the value itself with a controlled term.
E.g. here it would be

1 = `<Social Anxiety Disorder>`

### Current Behavior

I just see the row name in the .tsv. Probably because the data type of the row is numeric
(int64 in this case)

### Error message

_No response_

### Environment

_No response_

### How to reproduce

_No response_

### Anything else?

This came up in https://github.com/neurobagel/bulk_annotations/issues/10

So one fix would be to handle it as a categorical variable on account of the Levels key in the data dictionary. If the data dictionary was incomplete or missing, we would have to find another way to address this.

### PR DESCRIPTION
**Pull request by Remi-Gau**
Fixes #11  

Changes proposed in this pull request:

- do not skip annotation of column levels if those levels are already annotated in the participants.json
- avoid annotating levels twice (1 and "1" by casting all levels as string)
- do not skip yes/no binar columns

## Checklist

- [x] PR has an interpretable title with a prefix (`[ENH]`, `[BUG]`, `[DOC]`, `[INFRA]`, `[MAINT]`)
- [x] PR links to Github issue with mention `Closes #XXXX`
- [ ] Tests pass
- [x] Code is properly formatted

For new features:
- [ ] Tests have been added

For bug fixes:
- [ ] There is at least one test that would fail under the original bug conditions.

### CODE REVIEW

**Comment by Remi-Gau** [comment]
@surchs 
this should fix it

This adds a lot of lines to the TSV but most them should have annotations already.

I also decided to not "skip" columns that have a "yes/no" content because those should not add too many lines in the output.

Sorry if the diff of the TSV is not cleaner: I also sorted the levels before adding them to the TSV so that any eventual future diff will be cleaner.

**Comment by surchs** [comment]
Thanks so much @Remi-Gau for this super-fast PR. I'll take a look right away

File: list_participants_tsv_levels.py [hunk 1]
```diff
@@ -139,16 +145,11 @@ def list_levels(
...
-        and participants_dict.get(column)
-        and participants_dict[column].get("Levels")
-    ):
-        levels = participants_dict[column]["Levels"]
+    levels = get_levels_from_data_dict(participants_dict, column)
+    if levels:
         output = append_levels(output, levels, column, row_template)
 
-    actual_levels = list(participants[column].unique())
+    actual_levels = [str(x) for x in participants[column].unique()]
```
Comment (surchs) [inline list_participants_tsv_levels.py:152]: just out of curiosity: what's the reason for this change?
It'll work well for us, I think, because we cast everything to string. Just wanted to make sure I understand

**Review by surchs — APPROVED** [review]
Looks great. Thanks again @Remi-Gau. Good to go :cook:

File: list_participants_tsv_levels.py (hunk 1 above)
Comment (surchs) [reply list_participants_tsv_levels.py:152]: ah nvm, I read the above description in the PR. Missed this.

**Comment by Remi-Gau** [comment]
> super-fast PR

could have been quicker... I got distracted...
